code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -648,8 +649,9 @@
 
 /mob/living/critter/flock/drone/proc/add_resources(amount)
 	src.resources += amount
-	src.flock?.flockmind.tutorial?.PerformSilentAction(FLOCK_ACTION_GAIN_RESOURCES, src.resources)
-	src.flock?.resources_gained += amount
+	if (src.flock)
+		src.flock.flockmind.tutorial?.PerformSilentAction(FLOCK_ACTION_GAIN_RESOURCES, src.resources)
```

Comment:
```suggestion
		src.flock.flockmind.tutorial.PerformSilentAction(FLOCK_ACTION_GAIN_RESOURCES, src.resources)
```

Reply:
Tutorial will be null here when you're not in tutorial won't it?